x/blob/client/cli/payforblob.go

```diff
@@ -32,55 +32,79 @@ const (
 
 func CmdPayForBlob() *cobra.Command {
 	cmd := &cobra.Command{
-		Use: "PayForBlobs namespaceID blob",
+		Use: "PayForBlobs [path/to/blob.json]",
 		// This example command can be run in a new terminal after running single-node.sh
-		Example: "celestia-appd tx blob PayForBlobs 0x00010203040506070809 0x48656c6c6f2c20576f726c6421 \\\n" +
+		Example: "celestia-appd tx blob PayForBlobs path/to/blob.json \\\n" +
 			"\t--chain-id private \\\n" +
 			"\t--from validator \\\n" +
 			"\t--keyring-backend test \\\n" +
 			"\t--fees 21000utia \\\n" +
 			"\t--yes",
-		Short: "Pay for a data blob to be published to Celestia.",
-		Long: "Pay for a data blob to be published to Celestia.\n" +
-			"namespaceID is the user-specifiable portion of a version 0 namespace. It must be a hex encoded string of 10 bytes.\n" +
-			"blob must be a hex encoded string of any length.\n" +
-			// TODO: allow for more than one blob to be sumbmitted via the CLI
-			"This command currently only supports a single blob per invocation.\n",
+		Short: "Pay for a data blobs to be published to Celestia.",
+		Long: "Pay for a data blobs to be published to Celestia.\n" +
+			`Where blob.json contains: 
+
+		{
+			"Blobs": [
+				{
+					"namespaceId": "0x00010203040506070809",
+					"blob": "0x48656c6c6f2c20576f726c6421"
+				},
+				{
+					"namespaceId": "0x00010203040506070809",
+					"blob": "0x48656c6c6f2c20576f726c6421"
+				}
+			]
+		}
+
+		namespaceID is the user-specifiable portion of a version 0 namespace. It must be a hex encoded string of 10 bytes.\n
+		blob must be a hex encoded string of any length.\n
+		
+		`,
 		Aliases: []string{"PayForBlob"},
 		Args: func(cmd *cobra.Command, args []string) error {
-			if len(args) < 2 {
-				return fmt.Errorf("PayForBlobs requires two arguments: namespaceID and blob")
+			if len(args) < 1 {
+				return fmt.Errorf("PayForBlobs requires one arguments: path to blob.json")
```

Comment:
[blocking]
I think there's a way to avoid a breaking change, and we should pursue that. for example we could use a flag to indicate that its pointing to json file (or accepted piped input etc). we could even check if the input has the last five characters equal to ".json" and then proceed to try and parse a json

Reply:
added the check, thanks for the suggestion